products/billingbudget/api.yaml

```diff
@@ -0,0 +1,197 @@
+                  specified currently.
+                item_type: Api::Type::String
+              - !ruby/object:Api::Type::Enum
+                name: creditTypesTreatment
+                description: |
+                  Specifies how credits should be treated when determining spend
+                  for threshold calculations.
+                default_value: :INCLUDE_ALL_CREDITS
+                values:
+                  - :CREDIT_TYPES_TREATMENT_UNSPECIFIED
+                  - :INCLUDE_ALL_CREDITS
+                  - :EXCLUDE_ALL_CREDITS
+              - !ruby/object:Api::Type::Array
+                name: services
+                description: |
+                  A set of services of the form services/{service_id},
+                  specifying that usage from only this set of services should be
+                  included in the budget. If omitted, the report will include
+                  usage for all the services. The service names are available
+                  through the Catalog API:
+                  https://cloud.google.com/billing/v1/how-tos/catalog-api.
+                item_type: Api::Type::String
+          - !ruby/object:Api::Type::NestedObject
+            name: amount
+            required: true
+            description: |
+              The budgeted amount for each usage period. Exactly one of
+              :specifiedAmount or :lastPeriodAmount must be set.
+            properties:
+              - !ruby/object:Api::Type::NestedObject
+                name: specifiedAmount
+                description: |
+                  A specified amount to use as the budget. currencyCode is
+                  optional. If specified, it must match the currency of the
+                  billing account. The currencyCode is provided on output.
+                properties:
+                  - !ruby/object:Api::Type::String
+                    name: currencyCode
+                    description: |
+                      The 3-letter currency code defined in ISO 4217.
+                  - !ruby/object:Api::Type::String
+                    name: units
+                    description: |
+                      The whole units of the amount. For example if currencyCode
+                      is "USD", then 1 unit is one US dollar.
+                  - !ruby/object:Api::Type::Integer
+                    name: nanos
+                    description: |
+                      Number of nano (10^-9) units of the amount.
+                      The value must be between -999,999,999 and +999,999,999
+                      inclusive. If units is positive, nanos must be positive or
+                      zero. If units is zero, nanos can be positive, zero, or
+                      negative. If units is negative, nanos must be negative or
+                      zero. For example $-1.75 is represented as units=-1 and
+                      nanos=-750,000,000.
+              - !ruby/object:Api::Type::NestedObject
```

Comment:
I don't think we have the capability in Terraform, let alone MM, to have a nested object with nothing  inside of it. I'd say let's hold off on allowing people to specify this field, and make `specifiedAmount` required until the API puts something inside the `LastPeriodAmount` object.